x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/alerts_context.tsx

```diff
@@ -24,22 +25,42 @@ import type { AlertsTableImperativeApi } from '@kbn/response-ops-alerts-table/ty
  */
 const AlertsContext = createContext<{
   alertsTableRef: RefObject<AlertsTableImperativeApi>;
+  /**
+   * anonymization switch state in local storage\
+   * if undefined, the spaceId is not retrievable and the switch is not shown
+   */
+  showAnonymizedValues?: boolean;
+  setShowAnonymizedValues: React.Dispatch<React.SetStateAction<boolean | undefined>>;
 } | null>(null);
 
 const AlertsContextProviderComponent = ({ children }: PropsWithChildren) => {
   const alertsTableRef = useRef<AlertsTableImperativeApi>(null);
-  const contextValue = useMemo(() => ({ alertsTableRef }), []);
-  return <AlertsContext.Provider value={contextValue}>{children}</AlertsContext.Provider>;
+  const spaceId = useSpaceId();
+  const [showAnonymizedValues = spaceId ? false : undefined, setShowAnonymizedValues] =
```

Comment:
If we are not rendering differently between false and undefined, we could just put `false` here.
```suggestion
  const [showAnonymizedValues = false, setShowAnonymizedValues] =
```

Reply:
we are, if its undefined i don't show the switch

Reply:
Hey @stephmilovic , this context actually can re-render the whole table ( which we do not need) if there are changes in context value and it was added by @umbopepato(@elastic/response-ops) as a temporary measure which should be removed soon.

IS there any other place we can add these `showAnonymizedValues` state?

Since it is using local storage. Why does it needs to be the part of `context`?  Can't we simply use the hook wherever we need it? I will recommend that.

Alternatively, 

- if we want to do it with context, Do you think it can be done directly in `ActionsCell` ( [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/detections/components/alert_summary/table/actions_cell.tsx)) without including table in the context so that surface area of this value is as small as possible.

- We can also add it in `AIForSOCDetailsProvider` ([here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/flyout/ai_for_soc/context.tsx#L61))


Please correct me if I missed something.

Reply:
I see Jatin's concern. I'm a bit worried if`showAnonymizedValues` in the context is always sync with local storage. It'd be safer to access straight from the local storage value when needed.

Reply:
removed from this section

Reply:
Thanks @stephmilovic  for the changes.